### PR TITLE
create a google benchmark with chaos-mesh

### PR DESCRIPTION
**Pull request by chDame**

### CODE REVIEW

File: benchmark/Makefile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# Camunda components will be installed into the following Kubernetes namespace
+export namespace ?= camunda
+# Helm release name
+export release ?= camunda
+# Helm chart coordinates for Camunda
+export chart ?= camunda/camunda-platform
+
+# This file is created during the make call which runs inside `$(helmProfilesDir)/ingress-aws`
+export chartValues ?= "$(root)/benchmark/camunda-values-benchmark.yaml"
```
[the remark this answers is not in the document]
Comment (chDame) [reply benchmark/Makefile]: done

File: benchmark/Makefile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+.PHONY: chaos-all
+chaos-all: all chaosmesh
+
+.PHONY: rebalance-partitions
+rebalance-partitions:
+	kubectl port-forward --namespace camunda $(kubectl get pod --namespace camunda --selector="app=camunda-platform,app.kubernetes.io/component=zeebe-gateway,app.kubernetes.io/instance=camunda,app.kubernetes.io/managed-by=Helm,app.kubernetes.io/name=zeebe-gateway,app.kubernetes.io/part-of=camunda-platform" --output jsonpath='{.items[0].metadata.name}') 5432:9600 &
+	curl -X POST http://localhost:5432/actuator/rebalance/
+	sleep 10
+	pkill -f "port-forward"
```
[the remark this answers is not in the document]
Comment (chDame) [reply benchmark/Makefile]: done

File: benchmark/benchmark.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      app: benchmark
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: benchmark
+    spec:
+      containers:
+      - name: benchmark
+        image: gcr.io/camunda-researchanddevelopment/falko-camunda-8-benchmark:0.0.1-SNAPSHOT
```
[the remark this answers is not in the document]
Comment (chDame) [reply benchmark/benchmark.yaml]: done

File: benchmark/camunda-values-benchmark.yaml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  pvcSize: 128Gi
+  pvcStorageClassName: ssd
+  logLevel: ERROR
+  env:
+    - name: ZEEBE_BROKER_EXECUTION_METRICS_EXPORTER_ENABLED
+      value: "true"
+    - name: ZEEBE_BROKER_EXPERIMENTAL_DISABLEEXPLICITRAFTFLUSH
+      value: "true"
+    - name: ZEEBE_BROKER_BACKPRESSURE_ENABLED
+      value: "true"
```
[the remark this answers is not in the document]
Comment (chDame) [reply benchmark/camunda-values-benchmark.yaml]: done

File: benchmark/camunda-values-benchmark.yaml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+      cpu: 15
+      memory: 50Gi
+    requests:
+      cpu: 15
+      memory: 50Gi
+  clusterSize: "12"
+  partitionCount: "80"
+  replicationFactor: "3"
+  cpuThreadCount: 15
+  ioThreadCount: 15
```
[the remark this answers is not in the document]
Comment (chDame) [reply benchmark/camunda-values-benchmark.yaml]: done

File: benchmark/chaosmesh.mk [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+.PHONY: clean-chaos-mesh
+clean-chaos:
+	-rm -rf chaos-mesh
+	-helm --namespace default uninstall chaos-mesh
+	
+	
+	
+	kubectl apply -f network-chaos.yaml -n default
+	-kubectl delete -f network-chaos.yaml -n default
```
Comment (falko) [inline benchmark/chaosmesh.mk:19]: to be deleted

File: google/benchmark/README.md [hunk 7]
```diff
@@ -0,0 +1,40 @@
+# Camunda 8 Helm Profile: Ingress NGINX for GCP with TLS Certificates
```
Comment (falko) [inline google/benchmark/README.md:1]: Let's update the title

File: google/benchmark/README.md [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+If you need to create a new GKE Cluster, run `make kube`. 
+
+Once you have a GKE Cluster, run `make` to do the following:
+
+1. Set up a Kubernetes letsencrypt certificate manager
+2. Install a Kubernetes Nginx Ingress Controller. A corresponding GCP Load Balancer is provisioned automatically
+3. Attempt to find the ip address of the Load Balancer. This ip address is then used generate a `camunda-values.yaml` file. 
+4. Helm is used to install Camunda 8 using the `camunda-values.yaml` file with the Load Balancer IP Address
+5. The ingress controller is annotated so that letsencrypt tls certificates are provisioned.
```
Comment (falko) [inline google/benchmark/README.md:34]: either delete or describe what the makefile does

**Review by falko — CHANGES_REQUESTED** [review]
Submodule chaos-mesh should be removed

**Comment by falko** [comment]
What is missing to get this merged?

**Comment by upgradingdave** [comment]
Hi @falko, @chDame, I glanced thru the changes and I also read @falko's comments and I think all that is remaining before we can merge is to:

- [ ]  refactor the chaos mesh related files into a separate `chaos-mesh` directory
- [ ] Update `google/benchmark/README` with a few small changes to make things more clear
- [ ] Add a `README.md` to the `benchmark` directory

I can possibly work on these this afternoon and over the weekend, let me know if you can think of anything else to clean up before merging?

**Comment by falko** [comment]
I have some improvements in Zeebe Tuner that might be worth merging back to the Helm Profiles. In general, I'd like to get Zeebe Tuner use the profiles as much as possible so that we don't have to maintain two benchmark profiles in the end. I could look into that on Monday

**Comment by falko** [comment]
> Hi @falko, @chDame, I glanced thru the changes and I also read @falko's comments and I think all that is remaining before we can merge is to:
> 
> * [ ]   refactor the chaos mesh related files into a separate `chaos-mesh` directory
> * [ ]  Update `google/benchmark/README` with a few small changes to make things more clear
> * [ ]  Add a `README.md` to the `benchmark` directory
> 
> I can possibly work on these this afternoon and over the weekend, let me know if you can think of anything else to clean up before merging?

@upgradingdave feel free to contribute all these things to the main branch. @chDame and I will merge the PR now.

**Review by falko — APPROVED** [review]

**Pull request merged**